frontend/src/frontend/code_generator/user_callgraph.py

```diff
@@ -70,7 +76,41 @@ def format_vars_definition(self) -> str:
         return self.format_code_block_body(self.global_vars_def)
 
     def format_code_block_body(self, code_block: blocks.CodeBlock) -> str:
-        return code_block.get_instructions()
+        # code block body contains either instructions or a prefetch
+        cbb = code_block.code_block_body
+        try:
+            result = cbb.get_instructions_if_set()
+        except TypeError:
+            result = self.format_code_prefetch_instruction(
+                cbb.get_prefetch_inst_if_set())
+        return result
+
+    def format_code_prefetch_instruction(
+            self, code_prefetch: blocks.CodePrefetchInst) -> str:
+        if code_prefetch.type == blocks.TargetType.FUNCTION:
+            target_symbol = self.function_call_signature_for(
+                code_prefetch.target_name)
+        elif code_prefetch.type == blocks.TargetType.CODE_BLOCK:
+            target_symbol = '&' + self.format_code_block_label(
+                self.code_blocks[code_prefetch.target_name])
+        prefetch_string = self._create_prefetch_aarch64_string(
+            code_prefetch.degree, target_symbol)
+        result = ('#ifdef ENABLE_CODE_PREFETCH\n'
+                  '#ifdef __aarch64__\n'
+                  f'{prefetch_string}'
+                  '#endif\n'
+                  '#endif\n')
+        return result
+
+    def _create_prefetch_aarch64_string(self, degree: int,
+                                        target_symbol: str) -> str:
+        prefetch = ['"PRFM PLIL1KEEP, [%0]\\n\\t"\n']
```

Comment:
Slightly curious what the tab at the end is for?

Reply:
I'm following convention as seen here: https://gcc.gnu.org/onlinedocs/gcc/Extended-Asm.html 6.47.2.2 Assembler Template